SampleObjC/sdk-ios-objc/OCViewController.m

```diff
@@ -0,0 +1,109 @@
+//
+//  ViewController.m
+//  sdk-ios-objc
+//
+//  Created by Simon Hamadene on 16/10/2018.
+//  Copyright © 2018 Yoti Limited. All rights reserved.
+//
+
+#import "OCViewController.h"
+#import <YotiButtonSDK/YotiButtonSDK.h>
+#import "OCProfileViewController.h"
+
+@interface OCViewController () <YTBSDKDelegate, YTBBackendDelegate>
+@property (weak, nonatomic) IBOutlet YotiButton *rememberMe;
+@property (weak, nonatomic) IBOutlet YotiButton *selfieAuth;
+@end
+
+@implementation OCViewController {
+    NSArray *responseObject;
+}
+- (IBAction)buttonDidTouchUpInside:(YotiButton*)sender {
+    
+    NSString* useCaseID = sender.useCaseID;
+    NSError* error = nil;
+    
+    if (![useCaseID isEqual:@""]) {
+        [YotiSDK startScenarioForUseCaseID:useCaseID withDelegate:self error:&error];
+        
+        if (error != nil) {
+            NSLog(@"error : %@", error.description);
+        }
+    }
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    [self.rememberMe setTitle:@"OC RememberMe Scenario" forState: UIControlStateNormal ];
+    [self.selfieAuth setTitle:@"OC SelfieAuth Scenario" forState: UIControlStateNormal ];
+}
+
+-(void) moveToProfile {
+    [self performSegueWithIdentifier:@"moveToProfile" sender:self];
+}
+
+-(void) prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    
+    if ([[segue identifier] isEqualToString:@"moveToProfile"])
+    {
+        OCProfileViewController* profileViewController = [segue destinationViewController];
+        
+        for (id element in responseObject) {
```

Comment:
why not making the type specific?

Reply:
In the responseObject there are different types. the generic type works fine as well.

Reply:
Isn't that you are casting them all to `NSString*` in the loop ?

Reply:
the structure of the responseObject has been changed to make these type specific.